presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/clientproviders/RetryDriverExecutionMBean.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.connector.thrift.clientproviders;
+
+import com.facebook.presto.connector.thrift.annotations.ForRetryDriver;
+import io.airlift.concurrent.ThreadPoolExecutorMBean;
+import org.weakref.jmx.Managed;
+import org.weakref.jmx.Nested;
+
+import javax.inject.Inject;
+
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ThreadPoolExecutor;
+
+import static java.util.Objects.requireNonNull;
+
+public class RetryDriverExecutionMBean
```

Comment:
Why is this class needed?

Reply:
This is to expose the metrics of the retry thread pool. Since the lifecycle of the pool is managed by DI, we want DI to handle the JMX metric for the pool as well.

Reply:
I'm not sure how useful it will be. Let's remove it in this version.